src/databricks/labs/ucx/account/workspaces.py

```diff
@@ -76,21 +92,104 @@ def sync_workspace_info(self, workspaces: list[Workspace] | None = None):
             except (PermissionDenied, NotFound, ValueError):
                 logger.warning(f"Failed to save workspace info for {ws.config.host}")
 
-    def create_account_level_groups(self, prompts: Prompts):
-        acc_groups = self._get_account_groups()
+    def create_account_level_groups(self, prompts: Prompts) -> None:
+        """
+        Create account level groups from workspace groups
+
+        The following approach is used:
+        Get all valid worskpace groups from all workspaces
+
+        For each group:
+            - Check if the group already exists in the account
+            - If it does not exist, check if it is a nested group (users are added directly)
+            - If its a nested group follow the same approach recursively
+            - If it is a regular group, create the group in the account and add all members to the group
+        """
+        account_groups = self._get_account_groups()
         workspace_ids = [workspace.workspace_id for workspace in self._workspaces()]
         if not workspace_ids:
             raise ValueError("The workspace ids provided are not found in the account, Please check and try again.")
-        all_valid_workspace_groups = self._get_valid_workspaces_groups(prompts, workspace_ids)
+        context = AccountGroupCreationContext(
+            valid_workspace_groups={},
+            created_groups=set(),
+            created_groups_details={},
+            renamed_groups={},
+            account_groups=account_groups,
+        )
```

Comment:
Once the defaults are all properly in place:
```suggestion
        context = AccountGroupCreationContext()
```